plugins/thunderbolt/fu-plugin-thunderbolt.c

```diff
@@ -195,21 +195,23 @@ fu_plugin_thunderbolt_is_native (GUdevDevice *udevice, gboolean *is_native, GErr
 {
 	g_autoptr(GFile) nvmem = NULL;
 	g_autoptr(GBytes) controller_fw = NULL;
-	gchar *content;
-	gsize length;
+	g_autoptr(GInputStream) istr = NULL;
 
 	nvmem = fu_plugin_thunderbolt_find_nvmem (udevice, TRUE, error);
 	if (nvmem == NULL)
 		return FALSE;
 
-	if (!g_file_load_contents (nvmem, NULL, &content, &length, NULL, error))
+	/* read 128 bytes, so two 64 byte reads over the Thunderbolt fabric */
+	istr = G_INPUT_STREAM (g_file_read (nvmem, NULL, error));
+	if (istr == NULL)
+		return FALSE;
+	controller_fw = g_input_stream_read_bytes (istr, 128, NULL, error);
+	if (controller_fw == NULL)
 		return FALSE;
 
-	controller_fw = g_bytes_new_take (content, length);
-
-	return fu_plugin_thunderbolt_controller_is_native (controller_fw,
-							   is_native,
-							   error);
+	return fu_thunderbolt_image_controller_is_native (controller_fw,
```

Comment:
For the record, I had a quick peak that the function does check the bounds when reading stuff and it is (as expected) all fine.